### DISCUSSION TITLE
RuntimeError: Input type (MPSFloatType) and weight type (torch.FloatTensor) should be the same.

### DISCUSSION THREAD

**Discussion by iam-dante** [mrdbourke/pytorch-deep-learning#791] [Q&A] [answered]
I have placed all the weghts and model in the same device still this same error so please dont answer me with `.to(device)` on anything. And yes am using Mac Silicion Chip M2 pro.

**Comment by iam-dante** [comment, marked as answer]
So after experiments don't run the summary code 
```py
from torchinfo import summary

summary(model=model,
        input_size=(1,3,224,224),
        col_names=["input_size", "output_size", "num_params", "trainable"],
        col_width=20,
        row_settings=["var_names"])
```

The make the model back to cpu.

**Reply by justinkkgoh** [reply]
Am running M2 on my mac and skipping the summary also fixed this issue for me. Thanks for sharing

**Reply by ringdk** [reply]
This was exactly it, thank you @iam-dante
